File: 012.jarakirefactor/djeiva/oo/heranca_polimorf/ex1/erl/Tecnico.erl
%% Atributos:
%%		private float bonusSalarial;

%% Métodos:
%%	void exibeDados()

%% Atributos da superclasse:
%%		String nome;
%%		float salario;

%% Métodos das superclasses:
%%		void exibeDados()
%%		int getMatricula()

-module('Tecnico').
-export([exibeDados/1, getMatricula/1]).

getMatricula(ObjectID) ->
	'Assistente':getMatricula(ObjectID).

exibeDados(ObjectID) ->
	Scope = {exibeDados, 'Gerente'},

	Temp_Class1 = oo:super(ObjectID),
	Temp_Class1:exibeDados(ObjectID),

	io:format("~s~n", ["(Gerente)"]);

%% da superclasse Assistente
exibeDados(ObjectID) ->
	'Assistente':exibeDados(ObjectID).

%% da superclasse Funcionario
exibeDados(ObjectID) ->
	'Funcionario':exibeDados(ObjectID).

File: 012.jarakirefactor/djeiva/oo/heranca_polimorf/java/ex1/Tecnico.java
package ex1;

// bonus salarial
public class Tecnico extends Assistente {
	float bonusSalarial;

	public Tecnico(String nome, float salario, int matricula, float bonusSalarial) {
		super(nome, salario, matricula);
		this.bonusSalarial = bonusSalarial;
	}

	public void exibeDados() {
		super.exibeDados();
		System.out.println("Salario + bonus: " + (salario + bonusSalarial));
	}
}